docs/Populate.md

```diff
@@ -0,0 +1,118 @@
+# Populate Exchange
+
+`populate` is an exchange for auto-populating fields in your mutations.
+
+## How to use
+
+As with any exchange, add this to your client (before your cache).
+
+```tsx
+import { populateExchange } from '@urql/exchange-graphcache';
+
+const client = createClient({
+  // ...
+  exchanges: [dedupExchange, populateExchange, cacheExchange, fetchExchange],
+})
+```
+
+## Example usage
+
+Consider the following queries which have been requested
+
+
+```graphql
+# Query 1
+{
+  todos {
+    id
+    name
+  }
+} 
+
+# Query 2
+{
+  todos {
+    id
+    createdAt
+  }
+}
+```
+
+Now consider the following mutation which returns a todo
+
+```graphql
+# Without populate
+mutation addTodo(id: ID!) {
+  addTodo(id: $id) {
+    id        # To update Query 1 & 2
+    name      # To update Query 1
+    createdAt # To update Query 2
+  }
+}
+```
+
+By using the `populate` exchange, you no longer need to manually specify the selection set required to update your other queries.
+
+```graphql
+# With populate
+mutation addTodo(id: ID!) {
+  addTodo(id: $id) @populate
+}
+```
+
+> Note: The above two mutations produce an identical GraphQL request.
+
+### Choosing when to populate
+
+You may not want to populate your whole mutation response. In order to reduce your payload, pass populate lower in your query.
+
+```graphql
+mutation addTodo(id: ID!) {
+  addTodo(id: $id) {
+    id
+    user @populate
+  }
+}
+```
+
+### Using aliases
```

Comment:
Can we add a note here that hints at this maybe being supported automatically in the future? e.g.

> NOTE: Aliases may be added automatically by the `@populate` directive in the future. As a general rule, you should never rely on parts of your mutation response that have been populated automatically, as their shape may change unexpectedly.

Reply:
Yeah this is a good shout!

I don't know if we should mention aliases yet as we're not even sure what they would be used for. Maybe just a note about this evolving over time to support more configurability.

> As a general rule, you should never rely on parts of your mutation response that have been populated automatically, as their shape may change unexpectedly.

I'm not sure what this means...

Reply:
I think we should rephrase that second part, but we should discourage people from using any fields or selections in JS that populate has added automatically, if we want to add automatic aliasing.

Instead we should encourage them to add the fields they’re going to use in their own code explicitly, if possible. If we encourage this we enable more changes on our end in the future